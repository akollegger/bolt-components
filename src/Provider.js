import { Component } from "react";
import PropTypes from "prop-types";

class Provider extends Component {
  getChildContext() {
    return {
      driver: this.props.driver
    };
  }

  render() {
    return this.props.children;
  }
}

Provider.childContextTypes = {
  driver: PropTypes.object
};

Provider.propTypes = {
  driver: PropTypes.object.isRequired,
  children: PropTypes.node.isRequired
};

export default Provider;
