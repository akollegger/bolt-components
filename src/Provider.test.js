import React, { Component } from "react";
import PropTypes from "prop-types";
import TestRenderer from "react-test-renderer";

import Provider from "./Provider";

it("Passes driver in context to one child", () => {
  // Given
  const spy = jest.fn();
  const driver = { session: spy };
  class CheckContext extends Component {
    constructor(props, context) {
      super(props, context);
      context.driver.session(true);
    }
    render() {
      return null;
    }
  }
  CheckContext.contextTypes = {
    driver: PropTypes.object.isRequired
  };

  // When
  const out = TestRenderer.create(
    <Provider driver={driver}>
      <CheckContext />
    </Provider>
  );
  expect(out.toJSON()).toEqual(null);
  expect(spy).toHaveBeenCalledTimes(1);
  expect(spy).toHaveBeenLastCalledWith(true);
});

it("Passes driver in context to array of children", () => {
  // Given
  const spy = jest.fn();
  const driver = { session: spy };
  class CheckContext extends Component {
    constructor(props, context) {
      super(props, context);
      context.driver.session(true);
    }
    render() {
      return null;
    }
  }
  CheckContext.contextTypes = {
    driver: PropTypes.object.isRequired
  };

  const ArrayReturn = () => [
    <CheckContext key="1" />,
    <div key="2">Check this out</div>
  ];

  // When
  const out = TestRenderer.create(
    <Provider driver={driver}>
      <ArrayReturn />
    </Provider>
  );
  expect(out.toJSON()).toMatchSnapshot();
  expect(spy).toHaveBeenCalledTimes(1);
  expect(spy).toHaveBeenLastCalledWith(true);
});
