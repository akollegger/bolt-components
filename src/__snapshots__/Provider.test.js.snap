// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passes driver in context to array of children 1`] = `
<div>
  Check this out
</div>
`;
